datashader/utils.py

```diff
@@ -469,10 +479,127 @@ def mesh(vertices, simplices):
                                 'consider casting simplices to integers '
                                 'with ".astype(int)"')
 
-    assert len(vertices.columns) > 2 or simplices.values.shape[1] > 3, 'If no vertex weight column is provided, a triangle weight column is required.'
-
+    assert len(vertices.columns) > 2 or simplices.values.shape[1] > 3, \
+        'If no vertex weight column is provided, a triangle weight column is required.'
 
-    if isinstance(vertices, dd.DataFrame) and isinstance(simplices, dd.DataFrame):
+    if has_dask and isinstance(vertices, DaskDataFrame) and isinstance(simplices, DaskDataFrame):
         return _dd_mesh(vertices, simplices)
 
     return _pd_mesh(vertices, simplices)
+
+
+def pyspark_type_to_pandas_type(data_type):
```

Comment:
Is there a reason these functions are in utils.py, not pyspark.py?  Seems like it would be easier to maintain if all spark code were in one place, so people can see what needs changing.

Reply:
I thought it could go either way, but to isolate changes, I've moved everything into `pyspark.py`.